pkg/workflow/workflow.go

```diff
@@ -270,79 +290,88 @@ func (w *workflow) Trace() error {
 	return recorder.With(w.cli, w.app).Save("", data).Limit(10).Error()
 }
 
-func (w *workflow) GetBackoffWaitTime() time.Duration {
-	nextTime, ok := w.wfCtx.GetValueInMemory(wfTypes.ContextKeyNextExecuteTime)
-	if !ok {
-		return time.Second
+func (w *workflow) GetSuspendBackoffWaitTime() time.Duration {
+	if w.app.Status.Workflow == nil {
+		return 0
 	}
-	unix, ok := nextTime.(int64)
-	if !ok {
-		return time.Second
+	stepStatus := make(map[string]common.StepStatus)
+	for _, ss := range w.app.Status.Workflow.Steps {
+		setStepStatus(stepStatus, ss.StepStatus)
+		for _, sss := range ss.SubStepsStatus {
+			setStepStatus(stepStatus, sss.StepStatus)
+		}
 	}
-	next := time.Unix(unix, 0)
-	if next.After(time.Now()) {
-		return time.Until(next)
+	max := time.Duration(1<<63 - 1)
+	min := max
+	if w.app.Spec.Workflow != nil {
```

Comment:
we should check and return the nil case first to avoid complex.

```suggestion
	if w.app.Spec.Workflow == nil {
	    return 0
	}
```